docs/mechanical-brakes.md

```diff
@@ -0,0 +1,64 @@
+# Mechanical Brake
+
+Some systems employ mechanical brakes on motors as a safety feature. These brakes can also be engaged as a power-saving function if the motor is not moving, but still under load.
+
+ODrive supports the use of its GPIO pins to connect to external brake drive electronics.
+
+When the ODrive engages the drive electronics, the brake will be disabled. When the drive enters a fault or idle state, the brake will be re-engaged.
+
+---
+
+## Mechanical Brake Configuration
+Each axis supports one mechanical brake. The following properties are accessible through `odrivetool`:
+
+Name |  Type | Default
+--- | -- | -- 
+gpio_num | int | 0
+enabled | boolean | false
+is_active_low | boolean | true
+
+### gpio_num
+The GPIO pin number, according to the silkscreen labels on ODrive. Set with these commands:
+```
+<odrv>.<axis>.mechanical_brake.config.gpio_num = <1, 2, 3, 4, 5, 6, 7, 8>
+```
+After GPIO pin number is changed, you'll need to run `<odrv>.save_configuration()` and `<odrv>.reboot()` for changes to take effect.
+
+### enabled
+Enables/disables the operation of the mechanical brake.
+```
+<odrv>.<axis>.mechanical_brake.config.enabled = <True, False>
+```
+After mechanical brake is enabled or disabled, you'll need to run `<odrv>.save_configuration()` and `<odrv>.reboot()` for changes to take effect.
+
+### is_active_low
+Most safety braking systems are active low, e.g. when the power is off, the brake is on. If the system uses brake drive electronics which use active high logic, flip this bit then reconsider the safety implications of your design...
+
+### Example
+
+Let's say we're hacking away on an old ABB robotic arm. We've wired a 24V brake drive circuit triggered by GPIO5. When GPIO5 is driven, it will release the brakes on the axis we're moving.
+
+```
+<odrv>.<axis>.mechanical_brake.config.gpio_num = 5
+<odrv>.<axis>.mechanical_brake.config.enabled = True
```

Comment:
Please add: `<odrv>.config.gpio5_mode = GPIO_MODE_MECH_BRAKE`.

Reply:
Ahh, this is what I was trying to avoid. It's a bit of a bummer to make the user have to manually configure so many settings for each feature but I see what you're going for.

Reply:
Yes I agree so the plan is to have this sort of configuration done by the GUI eventually. Think of a picture of the ODrive where you can click on the GPIOs and select "Mechanical Brake" or drag a wire to a Mechanical Brake block. Or some kind of step-by-step setup wizard.

For this reason we're aiming for a low-level machine-to-machine API rather than a machine-to-user API that does things behind the back of the host.